objectherkenning_openbare_ruimte/databricks_pipelines/common/table_manager.py

```diff
@@ -0,0 +1,182 @@
+from datetime import datetime
+
+from pyspark.sql import DataFrame, SparkSession
+from pyspark.sql.types import StructType
+
+
+class TableManager:
+    def __init__(self, spark: SparkSession, catalog: str, schema: str):
+        self.spark = spark
+        self.catalog = catalog
+        self.schema = schema
+
+    def update_status(
+        self, table_name: str, job_process_time: datetime, exclude_ids=[]
+    ):
+        count_pending_query = f"""
+        SELECT COUNT(*) as pending_count
+        FROM {self.catalog}.{self.schema}.{table_name}
+        WHERE status = 'Pending'
+        """  # nosec
+        total_pending_before = self.spark.sql(count_pending_query).collect()[0][
+            "pending_count"
+        ]
+
+        update_query = f"""
+        UPDATE {self.catalog}.{self.schema}.{table_name}
+        SET status = 'Processed', processed_at = '{job_process_time}'
+        WHERE status = 'Pending'
+        """  # nosec
+        if exclude_ids:
+            exclude_ids_str = ", ".join(map(str, exclude_ids))
+            update_query += f" AND id NOT IN ({exclude_ids_str})"
+
+        self.spark.sql(update_query)
```

Comment:
Is it possible that returns the number of updated rows? In that case would it be useful to log this?

Reply:
i am doing this on line 41